### ISSUE TITLE
Random thoughts

### ISSUE THREAD

**Issue by angstsmurf** [i7/counterfeit-monkey#90]
This is just a list of unfinished thoughts concerning some potential areas of future work. Hopefully it will do more good here than on my hard drive.

**• Asking for directions**
Asking for directions is never actually useful, but I suppose it is a pretty nice easter egg. There is quite a bit of code dealing with it. Unfortunately it requires a lot of always-available quips that slow down conversations, as well as any other action while conversation is going on. In fact we somewhat arbitrarily remove all location-asking quips from play after act II for performance reasons. I have been considering adding a more general directions-asking: something that recognizes phrases like ASK WHERE TO FIND (a location) and HOW DO I GET TO (a location) (any location, or at least a reasonable subset). This usefulness of this is doubtful, however.

It would perhaps make more sense to be able to ASK ABOUT some of the locations on the map and get people’s opinion of the place rather than directions.

~~The ticket-taker has this reply when asked about Mediterranean food:~~

~~"The Babel Cafe is good," he says. "There are always a lot of university students over there, though, so it gets kind of crowded sometimes."~~

~~I suspect this will make many players ASK ABOUT BABEL CAFE, only to get the response ”That doesn't seem to be a topic of conversation at the moment.” Perhaps at least that one location-asking quip should be added. Or any kind of response, really.~~

~~**• Resetting tutorial**
Turning tutorial back on with TUTORIAL ON isn't really meaningful after leaving the starting area. Ideally, there should be a check in place to see if it is still meaningful to turn it on, and if not, inform the player about this and ask if they would want to restart the game with the tutorial switched on.~~

~~**• Disabling map**
Now that it is more-or-less playable on portable devices with tiny screens, the fact that the map covers most of the screen is probably more annoying than any remaining performance problems.~~

~~**• Thanking**
We can thank some people now, but not all of them, and only after the game decides we have a reason to. This is a bit arbitrary and subjective. Perhaps it would be better to allow the player to always thank anybody, but only give an interesting response in the cases we have currently implemented.~~

**• Highlighting**
This feature is supposed to help the player pick out the implemented things in room descriptions and other text, but I have not really checked how consistent it is. What should the guiding principle be? Is it really a good thing that scenery stuff that is not meant to be interacted with is highlighted?

**• Hints**
This is basically an undocumented  feature at the moment. Typing HINT ABOUT or THINK ABOUT something will give you a hint for some puzzles. But like the author mentions in the source, it would be a lot of work to give proper hints for the many multi-solution puzzles in the game. I’d say we’re about halfway there, though, so if someone is interested in working on it, please go ahead!

~~**• Floor/Ground**
We track whether a room is indoors or outdoors, but many messages use ”ground” or ”floor” seemingly arbitrarily. It shouldn’t be too hard to make this word conditional on the kind of room we are in.~~

**• Boar attack**
There are a couple of things the player might try against the boar that are not accounted for, such as setting the trap or pit-trap (are these even possible to bring to the shore?), or creating the roc.

**• Sink drain and sink tap products**
Trying to create things like ink drain, ink tap, sin tap, sin rain should at least give some kind of response.

**• Pas**
We already implement a pa and papas, so the plural of pa should not pose too much of a problem.

**• Fake people fighting for their lives**
As mentioned in the author’s notes it seems a little unfair that the game ends when the player tries to gel the mechanic. But the questionable ethics of creating people and then destroying them again is hinted at in a couple of places throughout the game. It might be cool if this was acknowledged by having living creatures panic or fight back when we try to remove them from existence. Just the occasional random message, not every single time. Perhaps they will still be angry or wary if recreated later.

**Comment by angstsmurf** [comment]
Disabling the map was implemented in 6afcda5 .

**Comment by angstsmurf** [comment]
Restarting the tutorial mode was fixed in 57c041b, and in b2db17e a response was added if trying to turn it on after the starting area.

There is still room for improvement: for example, once the "Finished tutorial mode" achievement is recorded and saved in the monkeyac file, the game will default to tutorial off when starting until monkeyac is deleted. Should turning tutorial mode back on remove the achievement from the file as well? On a tangential note, should there be an option to reset all achievements?

**Comment by angstsmurf** [comment]
I tried adding a quip about the Babel Café in [this](https://github.com/angstsmurf/counterfeit-monkey/tree/babel-topic) branch, but it is not ideal. It makes us ask something like "Do you know if there is a café nearby?" rather than about Babel specifically, and it increases opcode use by about 260 000 for a full play-through.

**Comment by curiousdannii** [comment]
That's a lot of opcodes for adding an object - not even a rule!

**Comment by angstsmurf** [comment]
Yeah. I suppose there simply are a lot of places where all quips are iterated through, so that's just the cost of adding a quip. But it sure seems like a lot.

**Comment by angstsmurf** [comment]
I added a new quip for this in 2175c46. Hopefully the performance loss is offset by the thanking changes in 83cabd7. Also added floor-ground substitution in ca36e44.

**Comment by angstsmurf** [comment]
Closed in favor of #135 and #136.
